public/API/API_GetUserGameLeaderboards.php

```diff
@@ -0,0 +1,124 @@
+use App\Support\Rules\CtypeAlnum;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'i' => ['required', 'min:1'],
+    'u' => ['required', 'min:2', 'max:20', new CtypeAlnum()],
+    'o' => ['sometimes', 'integer', 'min:0', 'nullable'],
+    'c' => ['sometimes', 'integer', 'min:1', 'max:500', 'nullable'],
+]);
+
+$offset = $input['o'] ?? 0;
+$count = $input['c'] ?? 200;
+
+$user = User::firstWhere('User', request()->query('u'));
+if (!$user) {
+    return response()->json(['User not found'], 404);
+}
+
+$game = Game::firstWhere("ID", request()->query('i'));
+if (!$game) {
+    return response()->json(['Game not found'], 404);
+}
+
+if ($game->leaderboards()->count() === 0) {
+    return response()->json(['Game has no leaderboards'], 422);
+}
+
+$userLeaderboardEntriesCount = LeaderboardEntry::where('user_id', $user->id)
+    ->whereIn('leaderboard_id', function ($query) use ($game) {
+        $query->select('ID')
+              ->from('LeaderboardDef')
+              ->where('GameID', $game->ID);
+    })
+    ->whereNull('deleted_at')
+    ->count();
+
+if ($userLeaderboardEntriesCount === 0) {
+    return response()->json(['User has no leaderboards on this game'], 422);
+}
+
+$leaderboardEntries = LeaderboardEntry::select('leaderboard_entries.*')
+    ->addSelect([
+        'calculated_rank' => LeaderboardEntry::from('leaderboard_entries as entries_bis')
+            ->join('LeaderboardDef as leaderboardDefBis', 'entries_bis.leaderboard_id', '=', 'leaderboardDefBis.ID')
+            ->whereColumn('entries_bis.leaderboard_id', 'leaderboard_entries.leaderboard_id')
+            ->whereNull('entries_bis.deleted_at')
+            ->whereNull('leaderboardDefBis.deleted_at')
+            ->where(function ($query) {
+                $query->where(function ($q) {
+                    $q->where('leaderboardDefBis.LowerIsBetter', 1)
+                      ->whereColumn('entries_bis.score', '<', 'leaderboard_entries.score');
+                })->orWhere(function ($q) {
+                    $q->where('leaderboardDefBis.LowerIsBetter', 0)
+                      ->whereColumn('entries_bis.score', '>', 'leaderboard_entries.score');
+                });
+            })
+            ->selectRaw('COUNT(*) + 1'),
+    ])
+    ->join('LeaderboardDef', 'leaderboard_entries.leaderboard_id', '=', 'LeaderboardDef.ID')
+    ->where('LeaderboardDef.GameID', $game->ID)
+    ->where('leaderboard_entries.user_id', $user->id)
+    ->whereNull('leaderboard_entries.deleted_at')
+    ->whereNull('LeaderboardDef.deleted_at')
+    ->with('leaderboard')
+    ->orderBy('LeaderboardDef.ID', 'asc')
+    ->skip($offset)
+    ->take($count)
+    ->get();
+
+$results = [];
+foreach ($leaderboardEntries as $leaderboardEntry) {
+    $results[] = [
+        'ID' => $leaderboardEntry->leaderboard->ID,
+        'RankAsc' => $leaderboardEntry->leaderboard->LowerIsBetter ? 'false' : 'true',
```

Comment:
It seems like it may be better to just return `true` or `false` here rather than "true" or "false".

Reply:
I think this is the result of a documentation error. I suggested making this look like the `API_GetGameLeaderboards` response, which says `RankAsc` is a string: https://github.com/RetroAchievements/RAWeb/blob/e402065f76947ac4ca356339c8eb1cc86b43b025/public/API/API_GetGameLeaderboards.php#L13

But it looks like it was changed to a true boolean at some point (or was just misdocumented):
https://github.com/RetroAchievements/RAWeb/blob/e402065f76947ac4ca356339c8eb1cc86b43b025/public/API/API_GetGameLeaderboards.php#L73

Reply:
Then is a good opportunity to update the documentation on the GetGameLeaderboards API, I'll do that and apply a _boolval_ to this one too